media-sound/cadence/cadence-9999.ebuild

```diff
@@ -0,0 +1,86 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+PYTHON_COMPAT=( python3_{6,7,8} )
+inherit git-r3 python-single-r1 xdg desktop
+
+DESCRIPTION="Collection of tools useful for audio production"
+HOMEPAGE="http://kxstudio.linuxaudio.org/Applications:Cadence"
+EGIT_REPO_URI="https://github.com/falkTX/Cadence.git"
+KEYWORDS=""
+LICENSE="GPL-2"
+SLOT="0"
+
+IUSE="a2jmidid ladish -pulseaudio opengl"
+
+REQUIRED_USE="${PYTHON_REQUIRED_USE}"
+
+CDEPEND="
+	${PYTHON_DEPS}
+	$(python_gen_cond_dep '
+		dev-python/dbus-python[${PYTHON_MULTI_USEDEP}]
+		dev-python/PyQt5[dbus,gui,opengl?,svg,widgets,${PYTHON_MULTI_USEDEP}]
+	')
+	media-sound/jack2[dbus]
+	media-sound/jack_capture
+	a2jmidid? ( media-sound/a2jmidid[dbus] )
+	ladish? ( >=media-sound/ladish-9999 )
+	pulseaudio? ( media-sound/pulseaudio[jack] )
+"
+RDEPEND="${CDEPEND}"
+DEPEND="${RDEPEND}"
+
+src_prepare() {
+	sed -i -e "s/python3/${EPYTHON}/" \
+		data/cadence \
+		data/cadence-aloop-daemon \
+		data/cadence-jacksettings \
+		data/cadence-logs \
+		data/cadence-render \
+		data/cadence-session-start \
+		data/catarina \
+		data/catia \
+		data/claudia \
+		data/claudia-launcher || die "sed failed"
+
+	default
+}
+
+src_compile() {
+	myemakeargs=(PREFIX="${EPREFIX}/usr"
+		SKIP_STRIPPING=true
+	)
+
+	emake "${myemakeargs[@]}"
+}
+
+src_install() {
+	emake PREFIX="${EPREFIX}/usr" DESTDIR="${ED}" install
+
+	python_fix_shebang ${ED}
+
+	# Clean up stuff that shouldn't be installed
+	rm -rf "${ED}"/etc/X11/xinit/xinitrc.d/61cadence-session-inject
+	rm -rf "${ED}"/etc/xdg/autostart/cadence-session-start.desktop
+	rm -rf "${ED}"/usr/share/applications/*.desktop
+
+	if use !pulseaudio; then
+		rm -rf "${ED}"/usr/bin/cadence-pulse2{jack,loopback}
```

Comment:
I vastly prefer just listing things separately. That way grepping ebuilds for filenames always works and tbh I also don't really see the gain, it takes me more brain-time to process this than just the straight direct lines.

Reply:
I took this from the in-tree ebuild. It reduces the code size a little bit and of course makes it easier to get the differences between in-tree ebuild and audio-overlay ebuild. The latter was exactly the reason why I chose to reduce the diff: To be able to see with one glance what the differences are and to keep both ebuils up-to-date simultainously.

However, I don't have a really strong feeling about this. If you wish, I can revert it.

Reply:
I'd prefer to keep it as it was.

With regards to the diff to upstream: I'd prefer to foxus on trying to make it so the ebuild in this overlay is no longer necessary and you (and anyone else) can just use the ebuild in the gentoo tree.

But I don't expect LADISH to be added to the gentoo tree, both because it's pretty much unmaintained and because it currently requires Python 2.7 (there have been three recent commits, but there's more to do to make it work with Python 3).
Would your use case also be served with NSM (https://github.com/linuxaudio/new-session-manager, it's a fork of Non Session Manager) instead of LADISH? It's the session manager the majority of the Linux audio community has decided to standardize on. It might take a bit of time before it offers the same features and UX as LADISH, especially with regards to the GUIs but it's at least a sustainable way forward.

Reply:
https://github.com/falkTX/Cadence/issues/142#issuecomment-453788617